### ISSUE TITLE
Allow after request middleware

### ISSUE THREAD

**Issue by Barbapapazes** [nitrojs/nitro#1301]
### Describe the feature

Hello,

I was thinking about after middleware like in express.

For example, if I've a middleware like this:

```ts
export default eventHandler(() => {
	console.log('before')
	next()
	console.log('after')
})
```
And a route like this:
```ts
export default eventHandler(() => {
	console.log('route')
	return
})
```

I expect:

```txt
before
route
after
```

_This is not an API to implement but juste an example of what I've in mind_

https://github.com/nuxt/nuxt/issues/21522

### Additional information

- [ ] Would you be willing to help implement this feature?

**Comment by gnoeley** [comment]
I recently came across this issue and ended up with something like,
```typescript
// middleware/afterware.ts
export default eventHandler((event) => {
  const start = Date.now();
  event.res.once('close', () => {
    const ms = Date.now() - start;
    console.log(`Request took ${ms}ms`);
  });
});
```
This was based on the node middleware helper function approach [here in the h3 library](https://github.com/unjs/h3/blob/04343587b620d41ee084668ba26e8e26a219e3af/src/adapters/node.ts#L33).
I'm not sure on the correctness of this but it would be nice if this was somehow supported.

Just noticed a recent change https://github.com/unjs/h3/pull/482 that has added the ability to add before/after hooks to h3 though. Could this be used to implement this functionality in Nitro using unjs/hookable?

**Comment by pi0** [comment]
@gnoeley yes that's the plan! You would be able to use nitro app hooks to leverage them.

**Comment by gnoeley** [comment]
If more hooks are to be added to Nitro, is there a plan to document them more clearly either as a main-section in docs or under the "Configuration" section? To date, I've been grepping the code to find our what was supported. Looks like they are all listed under [Plugins->Examples](https://nitro.unjs.io/guide/plugins#examples) currently, but perhaps calling them out directly could be useful. It was a point of confusion for me personally at least 😅 Happy to raise a PR if that would be welcome?

**Comment by pi0** [comment]
Certainly! I will make sure to at least write clear usage in PR description. You are always more than welcome to contribute and improve docs and examples ❤️

**Comment by gnoeley** [comment]
@pi0 , I've opened a draft PR with some proposed documentation changes. I've had a guess at what the final API for the addition of the new hooks could be, happy to accept comments once that is finalized and I can update. Can have any further discussion in the PR.

**Comment by pi0x** [comment]
Supported in v3 — `defineMiddleware((event, next) => ...)` gives you a `next()` you can await, so code after it runs once the route handler has returned (verified: before / route / after). There is also a `response` runtime hook for the same purpose.

_Triaged with AI assistance._
